src/app/screens/Onboard/ConnectRaspiBlitz/index.tsx

```diff
@@ -0,0 +1,158 @@
+import TextField from "../../../components/Form/TextField";
+import CompanionDownloadInfo from "../../../components/CompanionDownloadInfo";
+
+const initialFormData = Object.freeze({
+  url: "",
+  macaroon: "",
+});
+
+export default function ConnectRaspiBlitz() {
+  const navigate = useNavigate();
+  const [formData, setFormData] = useState(initialFormData);
+  const [loading, setLoading] = useState(false);
+
+  function handleChange(event: React.ChangeEvent<HTMLInputElement>) {
+    setFormData({
+      ...formData,
+      [event.target.name]: event.target.value.trim(),
+    });
+  }
+
+  function getConnectorType() {
+    if (formData.url.match(/\.onion/i)) {
+      return "nativelnd";
+    }
+    // default to LND
+    return "lnd";
+  }
+
+  async function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+    setLoading(true);
+    const { url, macaroon } = formData;
+    const account = {
+      name: "RaspiBlitz",
+      config: {
+        macaroon,
+        url,
+      },
+      connector: getConnectorType(),
+    };
+
+    try {
+      let validation;
+      // TODO: for native connectors we currently skip the validation because it is too slow (booting up Tor etc.)
+      if (account.connector === "nativelnd") {
+        validation = { valid: true, error: "" };
+      } else {
+        validation = await utils.call("validateAccount", account);
+      }
+
+      if (validation.valid) {
+        const addResult = await utils.call("addAccount", account);
+        if (addResult.accountId) {
+          await utils.call("selectAccount", {
+            id: addResult.accountId,
+          });
+          navigate("/test-connection");
+        }
+      } else {
+        alert(`
+          Connection failed. Are your RaspiBlitz credentials correct? \n\n(${validation.error})`);
+      }
+    } catch (e) {
+      console.error(e);
+      let message =
+        "Connection failed. Are your RaspiBlitz credentials correct?";
+      if (e instanceof Error) {
+        message += `\n\n${e.message}`;
+      }
+      alert(message);
+    }
+    setLoading(false);
+  }
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <div className="relative mt-14 lg:flex space-x-8 bg-white dark:bg-gray-800 px-12 py-10">
+        <div className="lg:w-1/2">
+          <h1 className="text-2xl font-bold dark:text-white">
+            Connect to your RaspiBlitz node
+          </h1>
+          <p className="text-gray-500 mt-6 dark:text-gray-400">
+            You need your node URL and a macaroon with read and send permissions
+            (e.g. admin.macaroon).
+            <br />
+            <br />
+            <b>SSH</b> into your <b>RaspiBlitz</b>.<br />
+            Run the command <b>sudo cat /mnt/hdd/tor/lndrest8080/hostname</b>.
+            <br />
+            Copy your <b>.onion</b> address.
+            <br />
+            Add <b>https://</b> at the beginning and <b>:8080</b> at the end.
```

Comment:
we actually could easily do that dynamically. 
always add https:// and have a second port field.
on change we then save `https://${url}:${port}` as `url`.

the user does not need to manually add.

Reply:
Great idea bumi!